components/feature/app-links/src/main/java/mozilla/components/feature/app/links/AppLinkRedirect.kt

```diff
@@ -0,0 +1,24 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package mozilla.components.feature.app.links
+
+import android.content.Intent
+
+/**
+ * Data class for the external Intent or fallback URL a given URL encodes for.
+ */
+data class AppLinkRedirect(
+    val appIntent: Intent?,
+    val webUrl: String?,
+    val isFallback: Boolean
+) {
+    fun hasExternalApp() = appIntent != null
+
+    fun hasFallback() = webUrl != null && isFallback
+
+    fun isRedirect() = hasExternalApp() || hasFallback()
+}
```

Comment:
I wonder if in a future iteration (not this PR) we could put this `AppLinkRedirect` into `Session` instead of a flag? Code (like the feature) could then react based on this high-level data class instead of just the flag (e.g. `AppLinksFeature` listens to a Redirect object in the Session).